model/task_lifecycle.go

```diff
@@ -574,6 +619,15 @@ func MarkStart(taskId string) error {
 		}
 	}
 
+	// if this is part of a display task, update the display task's status
```

Comment:
This occurs several times. Can this logic be refactored out, possibly in a way that makes it harder for us to forget to use it for future additions to the task lifecycle?